server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestNodesHotThreadsAction.java

```diff
@@ -37,20 +37,46 @@
 
 import static org.elasticsearch.rest.RestRequest.Method.GET;
 
-
 public class RestNodesHotThreadsAction extends BaseRestHandler {
 
+    static final String formatDeprecatedMessageWithoutNodeID = "%s is a deprecated endpoint. Please use [/nodes/hot_threads] instead.";
+    static final String formatDeprecatedMessageWithNodeID = "%s is a deprecated endpoint. Please use [/nodes/{nodeId}/hot_threads] instead.";
+    static final String DEPRECATED_MESSAGE_CLUSTER_NODES_HOT_THREADS = String.format(formatDeprecatedMessageWithoutNodeID, "[/_cluster/nodes/hot_threads]"); 
+    static final String DEPRECATED_MESSAGE_CLUSTER_NODES_NODEID_HOT_THREADS = String.format(formatDeprecatedMessageWithNodeID, "[/_cluster/nodes/{nodeId}/hot_threads]");
+    static final String DEPRECATED_MESSAGE_CLUSTER_NODES_HOTTHREADS = String.format(formatDeprecatedMessageWithoutNodeID, "[/_cluster/nodes/hotthreads]"); 
+    static final String DEPRECATED_MESSAGE_CLUSTER_NODES_NODEID_HOTTHREADS = String.format(formatDeprecatedMessageWithNodeID, "[/_cluster/nodes/{nodeId}/hotthreads]"); 
+    static final String DEPRECATED_MESSAGE_NODES_HOTTHREADS = String.format(formatDeprecatedMessageWithoutNodeID, "[/_nodes/hotthreads]"); 
+    static final String DEPRECATED_MESSAGE_NODES_NODEID_HOTTHREADS = String.format(formatDeprecatedMessageWithNodeID, "[/_nodes/{nodeId}/hotthreads]");
```

Comment:
You'll need to use the locale version of `String#format` and pass `Locale.ROOT`.

Reply:
Okay!

Reply:
Thank you :)